src/weights.jl

```diff
@@ -250,6 +250,65 @@ eweights(t::AbstractVector, r::AbstractRange, λ::Real) =
 
 # NOTE: No variance correction is implemented for exponential weights
 
+struct UnitWeights{S<:Integer, T<:Real} <: AbstractWeights{S, T, V where V<:Vector{T}}
+    len::S
+end
+
+@doc """
+    UnitWeights{T}(s)
+
+Construct a `UnitWeights` vector with length `s` and weight elements of type `T`.
+All weight elements are identically one.
+""" UnitWeights
+
+eltype(wv::UnitWeights{S, T}) where {S, T} = T
+values(wv::UnitWeights{S, T}) where {S, T} = fill(one(T), length(wv))
+sum(wv::UnitWeights{S, T}) where {S, T} = convert(T, length(wv))
+isempty(wv::UnitWeights) = iszero(wv.len)
+length(wv::UnitWeights) = wv.len
+Base.size(wv::UnitWeights) = Tuple(length(wv))
+
+@propagate_inbounds function Base.getindex(wv::UnitWeights{S, T}, i::Integer) where {S, T}
+    @boundscheck checkbounds(wv,i)
+    one(T)
+end
+
+@propagate_inbounds function Base.getindex(wv::UnitWeights{S, T}, i::AbstractArray{<:Int}) where {S, T}
```

Comment:
Actually I realize this definition isn't the most natural: we'd better return a new weights vector of the same type. But that's a separate issue since other types do the same thing.